sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -336,13 +346,12 @@ class __SYCL_EXPORT SYCLMemObjT : public SYCLMemObjI {
   EventImplPtr MInteropEvent;
   // Context passed by user to interoperability constructor.
   ContextImplPtr MInteropContext;
-  // OpenCL's memory object handle passed by user to interoperability
+  // Native backend memory object handle passed by user to interoperability
   // constructor.
-  // TODO update this member to support other backends.
-  cl_mem MInteropMemObject;
+  RT::PiMem MInteropMemObject;
   // Indicates whether memory object is created using interoperability
   // constructor or not.
-  bool MOpenCLInterop;
+  bool MOwnNativeHandle;
```

Comment:
It doesn't look pretty that the "ownership" knowledge (relevant for L0 backend only) is not opaque at the greater SYCL RT level. It's certainly OK to have it in the specific interop constructor, but do we really need to keep it in the class?

Reply:
Depends on our decision on "keep" ownership semantics, whether we want it to do the copy back to the original location or not. IF we go with get_native() approach you suggested, then we don't need this data member.

Reply:
I've changed the code to obtain info about ownership from piMemGetInfo.

Reply:
OK, that's better I think.